packages/oas/src/index.ts

```diff
@@ -291,6 +292,12 @@ export default class Oas {
     processing: boolean;
   };
 
+  /**
+   * Cache for transformed components, used to improve performance when processing multiple
+   * operations on the same API document. Limited to 100 entries with LRU eviction.
+   */
+  private componentCache?: Map<OASDocument, ComponentsObject | false>;
```

Comment:
I'm a little confused on the shape of this `Map` now because its primary key is the API definition that the instance of `Oas` has. Won't this always just store one component object?